### PR TITLE
Bump minimum Python version to 3.8

### PR DESCRIPTION
**Pull request by michaeldiamant**
Follows #322 to bump minimum Python version to 3.8.  Takes changes from the prior reverted move to 3.10:  https://github.com/algorand/py-algorand-sdk/pull/313.

### CODE REVIEW

**Review by algochoi — APPROVED** [review]
Looks good - hope we can also cross check the readthedocs build somehow as well.

**Review by tzaffi — APPROVED** [review]
LGTM

**Review by winder — APPROVED** [review]

File: .readthedocs.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# .readthedocs.yaml
```
[the remark this answers is not in the document]
Comment (michaeldiamant) [reply .readthedocs.yaml:1]: @winder I don't feel strongly.  I copied a prior PR and minimized diffs to keep the review simple.  

Though I philosophically agree with minimizing inline commentary.  Removing it.

```suggestion
```

[the remark this answers is not in the document]
Comment (winder) [reply .readthedocs.yaml]: I was mostly trying to refer to the other comments, like how to specify nodejs versions or build a pdf.

**Review by brianolson — APPROVED** [review]

File: .readthedocs.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (michaeldiamant) [reply .readthedocs.yaml]: @winder I attempted to address the feedback via https://github.com/algorand/py-algorand-sdk/pull/323/commits/98bdee3c1cc9b1b3648237adc62d27e455e8c66b.

**Pull request merged**